### PR TITLE
Inference of Smeagol

### PR DESCRIPTION
**Pull request by CorentinSeznec**
Creation of a SmeagolInferDataset and a InferSample class from the model of Poesy.
Creation of a JSON smeagol_infer dedicated to inference.
Minor corrections to poesy file

### CODE REVIEW

File: py4cast/datasets/smeagol.py [hunk 1]
```diff
@@ -779,3 +793,130 @@ def domain_info(self) -> DomainInfo:
...
+                        domain=data["grid"]["domain"],
+                        geometry=data["grid"]["geometry"],
+                        var=var_file,
+                    ),
+                    **vard,
+                )
+                param_list.append(param)
+
+        inference_period = Period(**conf["periods"]["test"], name="infer")
+        ds = InferSmeagolDataset(
```
Comment (A669015) [inline py4cast/datasets/smeagol.py:910]: You put the dataset instantiation in the for loop for Poesy, shouldn't be the same here ?

Comment (CorentinSeznec) [reply py4cast/datasets/smeagol.py:910]: Mhh I don't think so ? The for loop is used to add all parameters to a list. The dataset's instantiation takes this list as an input. For me Poesy and Smeagol are both instantiate dataset outside the loop.

**Review by flyIchtus — APPROVED** [review]

File: py4cast/datasets/poesy.py [hunk 2]
```diff
@@ -902,19 +899,20 @@ def from_json(
...
-            param_list,
-            PoesySettings(
-                members=members,
-                term=term,
-                num_input_steps=num_input_steps,
-                num_output_steps=0,
-                num_inference_pred_steps=config_override["num_inference_pred_steps"],
-            ),
-        )
+            inference_period = Period(**conf["periods"]["test"], name="infer")
```
Comment (A669015) [inline py4cast/datasets/poesy.py:902]: Here the code as been indented, pushing the dataset instantiation in the for loop (cf my other comment)

File: py4cast/datasets/smeagol.py (hunk 1 above)
Comment (A669015) [reply py4cast/datasets/smeagol.py:910]: Ok for me, but I mean there is a difference between the for loop in `poesy.py` (instantiation is in the _for_ loop) and `smeagol.py` file (here instantiation is after the _for_ loop). So please fix the mis-indentation in `poesy.py`

**Review by A669015 — CHANGES_REQUESTED** [review]

File: py4cast/datasets/smeagol.py (hunk 1 above)
Comment (CorentinSeznec) [reply py4cast/datasets/smeagol.py:910]: Oh okay! thanks for noticing it. It's fixed.

**Review by A669015 — APPROVED** [review]

**Pull request merged**